conference.js

```diff
@@ -600,76 +600,67 @@ export default {
 
                     return [];
                 });
-        } else if (!requestedAudio && !requestedVideo) {
-            // Resolve with no tracks
-            tryCreateLocalTracks = Promise.resolve([]);
-        } else {
+        } else if (requestedAudio || requestedVideo) {
             tryCreateLocalTracks = createLocalTracksF({
                 devices: initialDevices,
                 timeout,
                 firePermissionPromptIsShownEvent: true
             })
-                .catch(err => {
-                    if (requestedAudio && requestedVideo) {
-
-                        // Try audio only...
-                        errors.audioAndVideoError = err;
+            .catch(async error => {
+                if (error.name === JitsiTrackErrors.TIMEOUT) {
```

Comment:
Why did you removed the electron check?

Reply:
Even on electron, I didn't understand the purpose of retrying them here when the first gUM call has timed out. Maybe you know ?

Reply:
So the logic was for a normal browser we assume timeout is because the user haven't clicked on the prompt for permission. But on Electron there won't be permission prompt at all so we assume GUM is stuck and will never resolve.We actually had a lot of reports about this on Electron at some point.

IMHO it is good to keep the check.

Reply:
and executing gUM the second time when it was stuck on electron the first time around seems to fix it ? If thats the case, I don't mind bringing it back.